service/src/main/kotlin/fi/espoo/evaka/daycare/service/CaretakerService.kt

```diff
@@ -30,25 +26,24 @@ class CaretakerService(
             ORDER BY start_date DESC
             """.trimIndent()
 
-        return jdbc.query(
-            sql,
-            mapOf("groupId" to groupId)
-        ) { rs, _ ->
-            CaretakerAmount(
-                id = rs.getUUID("id"),
-                groupId = rs.getUUID("group_id"),
-                startDate = rs.getDate("start_date").toLocalDate(),
-                endDate = rs.getDate("end_date").toLocalDate().takeIf { it.isBefore(PGConstants.infinity) },
-                amount = rs.getDouble("amount")
-            )
-        }
+        return h.createQuery(sql)
+            .bind("groupId", groupId)
+            .map { rs, _ ->
+                CaretakerAmount(
+                    id = rs.getUUID("id"),
+                    groupId = rs.getUUID("group_id"),
+                    startDate = rs.getDate("start_date").toLocalDate(),
+                    endDate = rs.getDate("end_date").toLocalDate().takeIf { it.isBefore(PGConstants.infinity) },
+                    amount = rs.getDouble("amount")
+                )
+            }
+            .toList()
     }
 
-    @Transactional
-    fun insert(groupId: UUID, startDate: LocalDate, endDate: LocalDate?, amount: Double) {
+    fun insert(h: Handle, groupId: UUID, startDate: LocalDate, endDate: LocalDate?, amount: Double) {
         if (endDate != null && endDate.isBefore(startDate)) throw BadRequest("End date cannot be before start")
 
-        endPreviousRow(groupId, startDate)
+        endPreviousRow(h, groupId, startDate)
```

Comment:
Should this endPreviousRow and the later update be inside h.transaction to ensure there is a transaction when touching these two?

Reply:
Discussed this with the team, this is ok